zrml/parimutuel/src/lib.rs

```diff
@@ -405,10 +446,27 @@ mod pallet {
 
             T::AssetManager::withdraw(winning_asset, &who, withdrawn_asset_balance)?;
 
-            let remaining_bal = T::AssetManager::free_balance(market.base_asset, &pot_account);
+            let remaining_bal =
+                T::AssetManager::free_balance(market.base_asset.into(), &pot_account);
             let base_asset_payoff = payoff.min(remaining_bal);
 
-            T::AssetManager::transfer(market.base_asset, &pot_account, &who, base_asset_payoff)?;
+            T::AssetManager::transfer(
+                market.base_asset.into(),
+                &pot_account,
+                &who,
+                base_asset_payoff,
+            )?;
+
+            if outcome_total == winning_balance {
+                let destroy_result = T::AssetDestroyer::managed_destroy(winning_asset, None);
+                unreachable_non_terminating!(
+                    destroy_result.is_ok(),
+                    LOG_TARGET,
+                    "Can't destroy winning outcome asset {:?}: {:?}",
+                    winning_asset,
+                    destroy_result.err()
+                );
+            }
```

Comment:
I don't see another simpler solution. When the last winning tokens are exchanged for a reward, the asset has to be destroyed. The only alternative I see would be to automate redemption (automatically exchange all winning outcomes for rewards) and destroy the winning asset afterwards, which is obviously much more complex.
I think the additional effort for adding an asset to the destruction queue is also neglectable, the minor overpayment can be disregarded, especially since interacting with the network is insanely cheap anyways.

Reply:
Yeah, I don't see a better solution either.